CMakeLists.txt

```diff
@@ -459,22 +458,42 @@ if(ENABLE_CONAN)
   set(CONAN_BZIP2_VERSION 1.0.8)
   set(CONAN_EXPAT_VERSION 2.2.10)
   set(CONAN_LUA_VERSION 5.4.4)
-  set(CONAN_TBB_VERSION 2020.3)
+  set(CONAN_TBB_VERSION 2021.3.0)
   
   set(CONAN_SYSTEM_INCLUDES ON)
-  conan_cmake_run(
-    REQUIRES 
-    boost/${CONAN_BOOST_VERSION}
-    bzip2/${CONAN_BZIP2_VERSION}
-    expat/${CONAN_EXPAT_VERSION}
-    lua/${CONAN_LUA_VERSION}
-    tbb/${CONAN_TBB_VERSION}
-    BASIC_SETUP 
-    BUILD missing 
-    GENERATORS cmake_find_package
-    KEEP_RPATHS
-    NO_OUTPUT_DIRS
-  )
+
+
+  # explicitly say Conan to use x86 dependencies if build for x86 platforms (https://github.com/conan-io/cmake-conan/issues/141)
+  if(NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
+    conan_cmake_run(
+      REQUIRES 
+      boost/${CONAN_BOOST_VERSION}
+      bzip2/${CONAN_BZIP2_VERSION}
+      expat/${CONAN_EXPAT_VERSION}
+      lua/${CONAN_LUA_VERSION}
+      onetbb/${CONAN_TBB_VERSION}
+      BASIC_SETUP 
+      BUILD missing 
+      GENERATORS cmake_find_package
+      KEEP_RPATHS
+      NO_OUTPUT_DIRS
+      ARCH x86
```

Comment:
I am not CMake pro, so not sure how to reduce copy-paste here(the difference between branches is only this `ARCH x86` line)